typing/errortrace.mli

```diff
@@ -22,8 +22,16 @@ type position = First | Second
 val swap_position : position -> position
 val print_pos : Format.formatter -> position -> unit
 
-type desc = { t: type_expr; expanded: type_expr option }
-type 'a diff = { got: 'a; expected: 'a}
+type expanded_type = { ty: type_expr; expanded: type_expr }
+
+(** [unexpanded_type ty] creates an [expanded_type] whose expansion is also
+    [ty].  Usually, you want [Ctype.expand_type] instead, since the expansion
+    carries useful information; however, in certain circumstances, the error is
+    about the expansion of the type, meaning that actually performing the
+    expansion produces more confusing or inaccurate output. *)
+val unexpanded_type : type_expr -> expanded_type
```

Comment:
`trivial_expansion` was a great name, but I left `unexpanded_diff` alone because that seemed like the clearest name

Reply:
The only use of `unexpanded_diff` is well documented, thus the name this seems perfectly fine. A more self-documenting name might be `brittle_diff` or `fragile_diff` to explain that expanding the `diff` would break the error message.